docs/user-guide/gateway.md

```diff
@@ -0,0 +1,121 @@
+# Gateway
+
+## Gateway Connections
+
+The Gateway provides network services like: peering, NAT, and firewall for
+traffic entering and exiting the fabric, as well as traffic between VPCs. The
+Gateway is connected to the fabric spines, a minimum of two connections is
+required. The Gateway has its own ASN to use in the BGP routing protocol.
+Connections between the fabric and the outside world should use the Gateway.
+For example a data center WAN connection should meet the fabric at the Gateway.
+
+
+## Gateway Peering
+
+[VPC Peerings](vpcs.md#vpcpeering) provide VPC to VPC connectivity by way of the switches in the fabric.
+These peerings use route leaking inside the switch fabric to allow on VPC to communicate with another VPC.
+While this works well for many types of connections, especially those that need the full cut-through bandwidth of the fabric, the switch fabric is limited in the functionality in can provide with regard to the peering.
+For example, any peering that requires network address translation (NAT), or firewalling, and other functionality beyond the capabilities of the switch fabric cannot use a standard VPC peering.
+
+For such cases, Hedgehog provides a Gateway Peering feature in fabrics that have attached gateways.
+A Gateway Peering provides connectivity between VPCs (or externals) but causes the traffic to flow through a gateway node which can provide advanced services such as NAT and firewall.
+The downside is that the peering is limited in bandwidth to the capacity of the gateway node(s) instead of the full cut-through bandwidth of the fabric.
+
+
+### Simple Gateway Peering Between VPCs
+
+
+```{.yaml .annotate linenums="1" title="gw-peer.yaml"}
+apiVersion: gateway.githedgehog.com/v1alpha1
+kind: Peering
+metadata:
+  name: vpc-1--vpc-2
+  namespace: default
+spec:
+  peering:
+    vpc-1:
+      expose:
+        - ips:
+          - cidr: 10.0.0.0/24 # Expose all IP address in the 10.0.0.0/24 CIDR block to vpc-2
+    vpc-2:
+      expose:
+        - ips:
+          - cidr: 192.168.0.0/16 # Expose all IP addresses in the 192.168.0.0/16 CIDR block to vpc-1
+```
+
+A simple peering with no services deployed between the VPCs. This traffic will
+transit the gateway node.
+
+### Gateway Peering with stateless NAT
+
+```{.yaml .annotate linenums="1" title="sl-gw-peer.yaml"}
```

Comment:
@qmonnet, when you are able, can you confirm the yaml and the description are accurate? I think there is a room for users to be confused with my description. I want them to know which addresses from 10.0.1.0 will be NAT'd to which addresses in 10.11.11.0

Reply:
Yes looks good, thanks! But see my notes on the comments.